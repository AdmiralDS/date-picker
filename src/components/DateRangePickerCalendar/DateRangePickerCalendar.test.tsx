import { render } from '@testing-library/react';
import { ThemeProvider } from 'styled-components';

import { LIGHT_THEME } from '@admiral-ds/react-ui';
import { DateRangePickerCalendar } from '@admiral-ds/date-picker';

describe('DateRangePickerCalendar', () => {
  test('should render component', () => {
    const { container } = render(
      <ThemeProvider theme={LIGHT_THEME}>
        <DateRangePickerCalendar />
      </ThemeProvider>,
    );
    expect(container).toMatchSnapshot();
  });

  test('should render today date', () => {
    const wrapper = render(
      <ThemeProvider theme={LIGHT_THEME}>
        <DateRangePickerCalendar />
      </ThemeProvider>,
    );
    expect(wrapper).toBeTruthy();
    const currentDateCell = wrapper.container.querySelector('[data-is-current-day-cell]');
    expect(currentDateCell?.getAttribute('data-value')).toBe(new Date().toISOString().substring(0, 10));
  });
});
