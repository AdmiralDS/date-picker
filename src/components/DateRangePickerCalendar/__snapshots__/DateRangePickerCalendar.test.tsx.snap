// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DateRangePickerCalendar should render component 1`] = `
.c5 {
  position: absolute;
  top: 0;
  left: 0;
  height: 100%;
}

.c5>svg {
  height: 24px;
  width: 24px;
}

.c4 {
  width: 32px;
  height: 32px;
  border-radius: 50%;
  background-color: transparent;
  pointer-events: none;
  position: absolute;
  left: 50%;
  top: 50%;
  transform: translate(-50%, -50%);
}

.c2 {
  position: relative;
  padding: 0;
  margin: 4px;
  box-sizing: border-box;
  border: none;
  background-color: transparent;
  appearance: none;
  flex: 0 0 auto;
  height: 24px;
  width: 24px;
  border-radius: 4px;
  overflow: visible;
  cursor: pointer;
}

.c2 >* {
  pointer-events: none;
}

.c2:disabled {
  cursor: default;
  pointer-events: none;
}

.c2:disabled *[fill^='#'] {
  fill: #B0B5BD;
}

.c2:focus-visible {
  outline-offset: 2px;
  outline: #0062FF solid 2px;
}

.c2:focus >.c3 {
  background-color: transparent;
}

.c2:hover >.c3 {
  background-color: #0000000D;
}

.c2:active >.c3 {
  background-color: #0000001F;
}

.c2:focus-visible >.c3 {
  background-color: transparent;
}

.c1 {
  box-sizing: border-box;
  width: 284PX;
  height: 32px;
  padding: 0 12px;
  display: flex;
  justify-content: space-between;
  background-color: #FFFFFF;
}

.c6 {
  display: flex;
}

.c7 {
  padding: 4px 8px;
  border-radius: 16px;
  cursor: pointer;
  color: #0062FF;
  font-family: 'VTB Group UI',sans-serif;
  font-style: normal;
  font-weight: 550;
  font-size: 16px;
  line-height: 24px;
  font-feature-settings: 'tnum' on,'lnum' on;
  text-rendering: geometricPrecision;
  background-color: #FFFFFF;
}

.c7:hover {
  background-color: #0000000D;
}

.c11 {
  box-sizing: border-box;
  display: flex;
  margin-bottom: 4px;
}

.c12 {
  box-sizing: border-box;
  width: 36px;
  height: 36px;
  text-align: center;
  padding: 7px 0;
  font-family: 'VTB Group UI',sans-serif;
  font-style: normal;
  font-weight: 400;
  font-size: 14px;
  line-height: 20px;
  font-feature-settings: 'tnum' on,'lnum' on;
  text-rendering: geometricPrecision;
  color: #23262D;
  background-color: #FFFFFF;
  border: 1px #FFFFFF solid;
  cursor: default;
}

.c13 {
  box-sizing: border-box;
  display: flex;
  flex-wrap: wrap;
  align-content: space-between;
  height: 236px;
}

.c9 {
  background-color: #FFFFFF;
  width: 252px;
  margin: 12px 16px 16px 16px;
  font-family: 'VTB Group UI',sans-serif;
  font-style: normal;
  font-weight: 400;
  font-size: 14px;
  line-height: 20px;
  font-feature-settings: 'tnum' on,'lnum' on;
  text-rendering: geometricPrecision;
}

.c14 {
  position: relative;
  width: 36px;
  height: 36px;
  cursor: pointer;
}

.c14>* {
  pointer-events: none;
}

.c14[data-disabled-cell='true'] {
  cursor: default;
}

.c14[data-hidden-cell='true'] {
  cursor: default;
}

.c25 {
  position: relative;
  width: 84px;
  height: 36px;
  cursor: pointer;
}

.c25>* {
  pointer-events: none;
}

.c25[data-disabled-cell='true'] {
  cursor: default;
}

.c25[data-hidden-cell='true'] {
  cursor: default;
}

.c31 {
  position: relative;
  width: 60px;
  height: 36px;
  cursor: pointer;
}

.c31>* {
  pointer-events: none;
}

.c31[data-disabled-cell='true'] {
  cursor: default;
}

.c31[data-hidden-cell='true'] {
  cursor: default;
}

.c15 {
  position: absolute;
  top: 0;
  width: 50%;
  height: 100%;
  background-color: transparent;
  left: 0;
  border-top-left-radius: 4px;
  border-bottom-left-radius: 4px;
}

.c18 {
  position: absolute;
  top: 0;
  width: 50%;
  height: 100%;
  background-color: transparent;
  left: 0;
}

.c16 {
  position: absolute;
  top: 0;
  width: 50%;
  height: 100%;
  background-color: transparent;
  right: 0;
}

.c19 {
  position: absolute;
  top: 0;
  width: 50%;
  height: 100%;
  background-color: transparent;
  right: 0;
  border-top-right-radius: 4px;
  border-bottom-right-radius: 4px;
}

.c17 {
  position: absolute;
  top: 0;
  left: 0;
  box-sizing: border-box;
  text-align: center;
  width: 100%;
  height: 100%;
  padding: 7px 0;
  font-family: 'VTB Group UI',sans-serif;
  font-style: normal;
  font-weight: 400;
  font-size: 14px;
  line-height: 20px;
  font-feature-settings: 'tnum' on,'lnum' on;
  text-rendering: geometricPrecision;
  color: #23262D;
  background-color: #FFFFFF;
  border: 1px #FFFFFF solid;
  border-radius: 50%;
  color: #FFFFFF;
}

.c20 {
  position: absolute;
  top: 0;
  left: 0;
  box-sizing: border-box;
  text-align: center;
  width: 100%;
  height: 100%;
  padding: 7px 0;
  font-family: 'VTB Group UI',sans-serif;
  font-style: normal;
  font-weight: 400;
  font-size: 14px;
  line-height: 20px;
  font-feature-settings: 'tnum' on,'lnum' on;
  text-rendering: geometricPrecision;
  color: #23262D;
  background-color: #FFFFFF;
  border: 1px #FFFFFF solid;
  border-radius: 50%;
}

.c21 {
  position: absolute;
  top: 0;
  left: 0;
  box-sizing: border-box;
  text-align: center;
  width: 100%;
  height: 100%;
  padding: 7px 0;
  font-family: 'VTB Group UI',sans-serif;
  font-style: normal;
  font-weight: 400;
  font-size: 14px;
  line-height: 20px;
  font-feature-settings: 'tnum' on,'lnum' on;
  text-rendering: geometricPrecision;
  color: #23262D;
  background-color: #FFFFFF;
  border: 1px #FFFFFF solid;
  border-radius: 50%;
  border-color: #23262D;
}

.c26 {
  position: absolute;
  top: 0;
  left: 0;
  box-sizing: border-box;
  text-align: center;
  width: 100%;
  height: 100%;
  padding: 7px 0;
  font-family: 'VTB Group UI',sans-serif;
  font-style: normal;
  font-weight: 400;
  font-size: 14px;
  line-height: 20px;
  font-feature-settings: 'tnum' on,'lnum' on;
  text-rendering: geometricPrecision;
  color: #23262D;
  background-color: #FFFFFF;
  border: 1px #FFFFFF solid;
  border-radius: 18px;
}

.c27 {
  position: absolute;
  top: 0;
  left: 0;
  box-sizing: border-box;
  text-align: center;
  width: 100%;
  height: 100%;
  padding: 7px 0;
  font-family: 'VTB Group UI',sans-serif;
  font-style: normal;
  font-weight: 400;
  font-size: 14px;
  line-height: 20px;
  font-feature-settings: 'tnum' on,'lnum' on;
  text-rendering: geometricPrecision;
  color: #23262D;
  background-color: #FFFFFF;
  border: 1px #FFFFFF solid;
  border-radius: 18px;
  border-color: #23262D;
}

.c24 {
  height: 240px;
  box-sizing: border-box;
  display: flex;
  flex-wrap: wrap;
  align-content: space-between;
}

.c22 {
  background-color: #FFFFFF;
  width: 252px;
  margin: 28px 16px 36px 16px;
  font-family: 'VTB Group UI',sans-serif;
  font-style: normal;
  font-weight: 400;
  font-size: 14px;
  line-height: 20px;
  font-feature-settings: 'tnum' on,'lnum' on;
  text-rendering: geometricPrecision;
}

.c30 {
  height: 244px;
  box-sizing: border-box;
  display: flex;
  flex-wrap: wrap;
  align-content: space-between;
}

.c28 {
  background-color: #FFFFFF;
  width: 240px;
  margin: 28px 22px 32px 22px;
  font-family: 'VTB Group UI',sans-serif;
  font-style: normal;
  font-weight: 400;
  font-size: 14px;
  line-height: 20px;
  font-feature-settings: 'tnum' on,'lnum' on;
  text-rendering: geometricPrecision;
}

.c0 {
  box-sizing: border-box;
  display: flex;
  flex-direction: column;
  align-items: center;
  align-content: space-between;
  padding-top: 20px;
  width: 284px;
  height: 356px;
  background-color: #FFFFFF;
  border-radius: 4px;
  box-shadow: 0px 3.2px 9px 0px rgba(0, 0, 0, 0.16),0px 0.6px 1.8px 0px rgba(0, 0, 0, 0.10),0px -1.5px 6px 0px rgba(0, 0, 0, 0.06);
}

.c8 {
  position: relative;
  width: 100%;
  height: 100%;
  padding: 0;
  margin: 0;
}

.c8>div {
  position: absolute;
  top: 0;
  left: 0;
}

.c10 {
  visibility: visible;
}

.c23 {
  visibility: hidden;
}

.c29 {
  visibility: hidden;
}

<div>
  <div
    class="c0"
  >
    <div
      class="c1"
    >
      <button
        class="c2"
        data-panel-target-type="left"
        type="button"
      >
        <div
          aria-hidden="true"
          class="c3 c4"
        />
        <div
          aria-hidden="true"
          class="c5"
        >
          <svg
            fill="none"
            focusable="false"
            viewBox="0 0 24 24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M14.508 18.754c.18 0 .35-.07.48-.21.24-.27.22-.68-.04-.92l-6.369-5.74 6.41-5.55a.65.65 0 0 0 .04-.92.658.658 0 0 0-.92-.04l-6.9 5.99c-.31.28-.31.76 0 1.04l6.86 6.18c.13.12.28.17.44.17"
              fill="#717681"
            />
          </svg>
        </div>
      </button>
      <div
        class="c6"
      >
        <div
          class="c7"
          data-panel-target-type="month"
        >
          Февраль
        </div>
        <div
          class="c7"
          data-panel-target-type="year"
        >
          2024
        </div>
      </div>
      <button
        class="c2"
        data-panel-target-type="right"
        type="button"
      >
        <div
          aria-hidden="true"
          class="c3 c4"
        />
        <div
          aria-hidden="true"
          class="c5"
        >
          <svg
            fill="none"
            focusable="false"
            viewBox="0 0 24 24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M9.474 18.754c-.18 0-.35-.07-.48-.21a.658.658 0 0 1 .04-.92l6.37-5.74-6.41-5.55a.65.65 0 0 1-.04-.92c.24-.26.65-.28.92-.04l6.9 5.99c.31.28.31.76 0 1.04l-6.86 6.18c-.13.12-.28.17-.44.17"
              fill="#717681"
            />
          </svg>
        </div>
      </button>
    </div>
    <div
      class="c8"
    >
      <div
        class="c9 c10"
        data-container-type="datesOfMonthWrapper"
      >
        <div
          class="c11"
        >
          <div
            class="c12"
            data-cell-type="dayCell"
            data-value="пн"
          >
            Пн
          </div>
          <div
            class="c12"
            data-cell-type="dayCell"
            data-value="вт"
          >
            Вт
          </div>
          <div
            class="c12"
            data-cell-type="dayCell"
            data-value="ср"
          >
            Ср
          </div>
          <div
            class="c12"
            data-cell-type="dayCell"
            data-value="чт"
          >
            Чт
          </div>
          <div
            class="c12"
            data-cell-type="dayCell"
            data-value="пт"
          >
            Пт
          </div>
          <div
            class="c12"
            data-cell-type="dayCell"
            data-value="сб"
          >
            Сб
          </div>
          <div
            class="c12"
            data-cell-type="dayCell"
            data-value="вс"
          >
            Вс
          </div>
        </div>
        <div
          class="c13"
          data-container-type="datesWrapper"
        >
          <div
            class="c14"
            data-cell-type="dateCell"
            data-hidden-cell="true"
            data-is-outside-month-date="true"
            data-is-start-of-week-cell="true"
            data-value="2024-01-29"
          >
            <div
              class="c15"
            />
            <div
              class="c16"
            />
            <div
              class="c17"
            >
              29
            </div>
          </div>
          <div
            class="c14"
            data-cell-type="dateCell"
            data-hidden-cell="true"
            data-is-outside-month-date="true"
            data-value="2024-01-30"
          >
            <div
              class="c18"
            />
            <div
              class="c16"
            />
            <div
              class="c17"
            >
              30
            </div>
          </div>
          <div
            class="c14"
            data-cell-type="dateCell"
            data-hidden-cell="true"
            data-is-end-of-week-cell="true"
            data-is-outside-month-date="true"
            data-value="2024-01-31"
          >
            <div
              class="c18"
            />
            <div
              class="c19"
            />
            <div
              class="c17"
            >
              31
            </div>
          </div>
          <div
            class="c14"
            data-cell-type="dateCell"
            data-is-start-of-week-cell="true"
            data-value="2024-02-01"
          >
            <div
              class="c15"
            />
            <div
              class="c16"
            />
            <div
              class="c20"
            >
              1
            </div>
          </div>
          <div
            class="c14"
            data-cell-type="dateCell"
            data-value="2024-02-02"
          >
            <div
              class="c18"
            />
            <div
              class="c16"
            />
            <div
              class="c20"
            >
              2
            </div>
          </div>
          <div
            class="c14"
            data-cell-type="dateCell"
            data-value="2024-02-03"
          >
            <div
              class="c18"
            />
            <div
              class="c16"
            />
            <div
              class="c20"
            >
              3
            </div>
          </div>
          <div
            class="c14"
            data-cell-type="dateCell"
            data-is-end-of-week-cell="true"
            data-value="2024-02-04"
          >
            <div
              class="c18"
            />
            <div
              class="c19"
            />
            <div
              class="c20"
            >
              4
            </div>
          </div>
          <div
            class="c14"
            data-cell-type="dateCell"
            data-is-start-of-week-cell="true"
            data-value="2024-02-05"
          >
            <div
              class="c15"
            />
            <div
              class="c16"
            />
            <div
              class="c20"
            >
              5
            </div>
          </div>
          <div
            class="c14"
            data-cell-type="dateCell"
            data-value="2024-02-06"
          >
            <div
              class="c18"
            />
            <div
              class="c16"
            />
            <div
              class="c20"
            >
              6
            </div>
          </div>
          <div
            class="c14"
            data-cell-type="dateCell"
            data-value="2024-02-07"
          >
            <div
              class="c18"
            />
            <div
              class="c16"
            />
            <div
              class="c20"
            >
              7
            </div>
          </div>
          <div
            class="c14"
            data-cell-type="dateCell"
            data-value="2024-02-08"
          >
            <div
              class="c18"
            />
            <div
              class="c16"
            />
            <div
              class="c20"
            >
              8
            </div>
          </div>
          <div
            class="c14"
            data-cell-type="dateCell"
            data-value="2024-02-09"
          >
            <div
              class="c18"
            />
            <div
              class="c16"
            />
            <div
              class="c20"
            >
              9
            </div>
          </div>
          <div
            class="c14"
            data-cell-type="dateCell"
            data-value="2024-02-10"
          >
            <div
              class="c18"
            />
            <div
              class="c16"
            />
            <div
              class="c20"
            >
              10
            </div>
          </div>
          <div
            class="c14"
            data-cell-type="dateCell"
            data-is-end-of-week-cell="true"
            data-value="2024-02-11"
          >
            <div
              class="c18"
            />
            <div
              class="c19"
            />
            <div
              class="c20"
            >
              11
            </div>
          </div>
          <div
            class="c14"
            data-cell-type="dateCell"
            data-is-start-of-week-cell="true"
            data-value="2024-02-12"
          >
            <div
              class="c15"
            />
            <div
              class="c16"
            />
            <div
              class="c20"
            >
              12
            </div>
          </div>
          <div
            class="c14"
            data-cell-type="dateCell"
            data-value="2024-02-13"
          >
            <div
              class="c18"
            />
            <div
              class="c16"
            />
            <div
              class="c20"
            >
              13
            </div>
          </div>
          <div
            class="c14"
            data-cell-type="dateCell"
            data-value="2024-02-14"
          >
            <div
              class="c18"
            />
            <div
              class="c16"
            />
            <div
              class="c20"
            >
              14
            </div>
          </div>
          <div
            class="c14"
            data-cell-type="dateCell"
            data-value="2024-02-15"
          >
            <div
              class="c18"
            />
            <div
              class="c16"
            />
            <div
              class="c20"
            >
              15
            </div>
          </div>
          <div
            class="c14"
            data-cell-type="dateCell"
            data-value="2024-02-16"
          >
            <div
              class="c18"
            />
            <div
              class="c16"
            />
            <div
              class="c20"
            >
              16
            </div>
          </div>
          <div
            class="c14"
            data-cell-type="dateCell"
            data-value="2024-02-17"
          >
            <div
              class="c18"
            />
            <div
              class="c16"
            />
            <div
              class="c20"
            >
              17
            </div>
          </div>
          <div
            class="c14"
            data-cell-type="dateCell"
            data-is-end-of-week-cell="true"
            data-value="2024-02-18"
          >
            <div
              class="c18"
            />
            <div
              class="c19"
            />
            <div
              class="c20"
            >
              18
            </div>
          </div>
          <div
            class="c14"
            data-cell-type="dateCell"
            data-is-start-of-week-cell="true"
            data-value="2024-02-19"
          >
            <div
              class="c15"
            />
            <div
              class="c16"
            />
            <div
              class="c20"
            >
              19
            </div>
          </div>
          <div
            class="c14"
            data-cell-type="dateCell"
            data-value="2024-02-20"
          >
            <div
              class="c18"
            />
            <div
              class="c16"
            />
            <div
              class="c20"
            >
              20
            </div>
          </div>
          <div
            class="c14"
            data-cell-type="dateCell"
            data-value="2024-02-21"
          >
            <div
              class="c18"
            />
            <div
              class="c16"
            />
            <div
              class="c20"
            >
              21
            </div>
          </div>
          <div
            class="c14"
            data-cell-type="dateCell"
            data-value="2024-02-22"
          >
            <div
              class="c18"
            />
            <div
              class="c16"
            />
            <div
              class="c20"
            >
              22
            </div>
          </div>
          <div
            class="c14"
            data-cell-type="dateCell"
            data-value="2024-02-23"
          >
            <div
              class="c18"
            />
            <div
              class="c16"
            />
            <div
              class="c20"
            >
              23
            </div>
          </div>
          <div
            class="c14"
            data-cell-type="dateCell"
            data-value="2024-02-24"
          >
            <div
              class="c18"
            />
            <div
              class="c16"
            />
            <div
              class="c20"
            >
              24
            </div>
          </div>
          <div
            class="c14"
            data-cell-type="dateCell"
            data-is-end-of-week-cell="true"
            data-value="2024-02-25"
          >
            <div
              class="c18"
            />
            <div
              class="c19"
            />
            <div
              class="c20"
            >
              25
            </div>
          </div>
          <div
            class="c14"
            data-cell-type="dateCell"
            data-is-current-day-cell="true"
            data-is-start-of-week-cell="true"
            data-value="2024-02-26"
          >
            <div
              class="c15"
            />
            <div
              class="c16"
            />
            <div
              class="c21"
            >
              26
            </div>
          </div>
          <div
            class="c14"
            data-cell-type="dateCell"
            data-value="2024-02-27"
          >
            <div
              class="c18"
            />
            <div
              class="c16"
            />
            <div
              class="c20"
            >
              27
            </div>
          </div>
          <div
            class="c14"
            data-cell-type="dateCell"
            data-value="2024-02-28"
          >
            <div
              class="c18"
            />
            <div
              class="c16"
            />
            <div
              class="c20"
            >
              28
            </div>
          </div>
          <div
            class="c14"
            data-cell-type="dateCell"
            data-is-end-of-week-cell="true"
            data-value="2024-02-29"
          >
            <div
              class="c18"
            />
            <div
              class="c19"
            />
            <div
              class="c20"
            >
              29
            </div>
          </div>
          <div
            class="c14"
            data-cell-type="dateCell"
            data-hidden-cell="true"
            data-is-outside-month-date="true"
            data-is-start-of-week-cell="true"
            data-value="2024-03-01"
          >
            <div
              class="c15"
            />
            <div
              class="c16"
            />
            <div
              class="c17"
            >
              1
            </div>
          </div>
          <div
            class="c14"
            data-cell-type="dateCell"
            data-hidden-cell="true"
            data-is-outside-month-date="true"
            data-value="2024-03-02"
          >
            <div
              class="c18"
            />
            <div
              class="c16"
            />
            <div
              class="c17"
            >
              2
            </div>
          </div>
          <div
            class="c14"
            data-cell-type="dateCell"
            data-hidden-cell="true"
            data-is-end-of-week-cell="true"
            data-is-outside-month-date="true"
            data-value="2024-03-03"
          >
            <div
              class="c18"
            />
            <div
              class="c19"
            />
            <div
              class="c17"
            >
              3
            </div>
          </div>
          <div
            class="c14"
            data-cell-type="dateCell"
            data-hidden-cell="true"
            data-is-outside-month-date="true"
            data-is-start-of-week-cell="true"
            data-value="2024-03-04"
          >
            <div
              class="c15"
            />
            <div
              class="c16"
            />
            <div
              class="c17"
            >
              4
            </div>
          </div>
          <div
            class="c14"
            data-cell-type="dateCell"
            data-hidden-cell="true"
            data-is-outside-month-date="true"
            data-value="2024-03-05"
          >
            <div
              class="c18"
            />
            <div
              class="c16"
            />
            <div
              class="c17"
            >
              5
            </div>
          </div>
          <div
            class="c14"
            data-cell-type="dateCell"
            data-hidden-cell="true"
            data-is-outside-month-date="true"
            data-value="2024-03-06"
          >
            <div
              class="c18"
            />
            <div
              class="c16"
            />
            <div
              class="c17"
            >
              6
            </div>
          </div>
          <div
            class="c14"
            data-cell-type="dateCell"
            data-hidden-cell="true"
            data-is-outside-month-date="true"
            data-value="2024-03-07"
          >
            <div
              class="c18"
            />
            <div
              class="c16"
            />
            <div
              class="c17"
            >
              7
            </div>
          </div>
          <div
            class="c14"
            data-cell-type="dateCell"
            data-hidden-cell="true"
            data-is-outside-month-date="true"
            data-value="2024-03-08"
          >
            <div
              class="c18"
            />
            <div
              class="c16"
            />
            <div
              class="c17"
            >
              8
            </div>
          </div>
          <div
            class="c14"
            data-cell-type="dateCell"
            data-hidden-cell="true"
            data-is-outside-month-date="true"
            data-value="2024-03-09"
          >
            <div
              class="c18"
            />
            <div
              class="c16"
            />
            <div
              class="c17"
            >
              9
            </div>
          </div>
          <div
            class="c14"
            data-cell-type="dateCell"
            data-hidden-cell="true"
            data-is-end-of-week-cell="true"
            data-is-outside-month-date="true"
            data-value="2024-03-10"
          >
            <div
              class="c18"
            />
            <div
              class="c19"
            />
            <div
              class="c17"
            >
              10
            </div>
          </div>
        </div>
      </div>
      <div
        class="c22 c23"
        data-container-type="monthsOfYearWrapper"
      >
        <div
          class="c24"
          data-container-type="monthsWrapper"
        >
          <div
            class="c25"
            data-cell-type="monthCell"
            data-value="Mon, 01 Jan 2024 09:00:00 GMT"
          >
            <div
              class="c18"
            />
            <div
              class="c16"
            />
            <div
              class="c26"
            >
              Январь
            </div>
          </div>
          <div
            class="c25"
            data-cell-type="monthCell"
            data-is-current-month="true"
            data-value="Thu, 01 Feb 2024 09:00:00 GMT"
          >
            <div
              class="c18"
            />
            <div
              class="c16"
            />
            <div
              class="c27"
            >
              Февраль
            </div>
          </div>
          <div
            class="c25"
            data-cell-type="monthCell"
            data-value="Fri, 01 Mar 2024 09:00:00 GMT"
          >
            <div
              class="c18"
            />
            <div
              class="c16"
            />
            <div
              class="c26"
            >
              Март
            </div>
          </div>
          <div
            class="c25"
            data-cell-type="monthCell"
            data-value="Mon, 01 Apr 2024 09:00:00 GMT"
          >
            <div
              class="c18"
            />
            <div
              class="c16"
            />
            <div
              class="c26"
            >
              Апрель
            </div>
          </div>
          <div
            class="c25"
            data-cell-type="monthCell"
            data-value="Wed, 01 May 2024 09:00:00 GMT"
          >
            <div
              class="c18"
            />
            <div
              class="c16"
            />
            <div
              class="c26"
            >
              Май
            </div>
          </div>
          <div
            class="c25"
            data-cell-type="monthCell"
            data-value="Sat, 01 Jun 2024 09:00:00 GMT"
          >
            <div
              class="c18"
            />
            <div
              class="c16"
            />
            <div
              class="c26"
            >
              Июнь
            </div>
          </div>
          <div
            class="c25"
            data-cell-type="monthCell"
            data-value="Mon, 01 Jul 2024 09:00:00 GMT"
          >
            <div
              class="c18"
            />
            <div
              class="c16"
            />
            <div
              class="c26"
            >
              Июль
            </div>
          </div>
          <div
            class="c25"
            data-cell-type="monthCell"
            data-value="Thu, 01 Aug 2024 09:00:00 GMT"
          >
            <div
              class="c18"
            />
            <div
              class="c16"
            />
            <div
              class="c26"
            >
              Август
            </div>
          </div>
          <div
            class="c25"
            data-cell-type="monthCell"
            data-value="Sun, 01 Sep 2024 09:00:00 GMT"
          >
            <div
              class="c18"
            />
            <div
              class="c16"
            />
            <div
              class="c26"
            >
              Сентябрь
            </div>
          </div>
          <div
            class="c25"
            data-cell-type="monthCell"
            data-value="Tue, 01 Oct 2024 09:00:00 GMT"
          >
            <div
              class="c18"
            />
            <div
              class="c16"
            />
            <div
              class="c26"
            >
              Октябрь
            </div>
          </div>
          <div
            class="c25"
            data-cell-type="monthCell"
            data-value="Fri, 01 Nov 2024 09:00:00 GMT"
          >
            <div
              class="c18"
            />
            <div
              class="c16"
            />
            <div
              class="c26"
            >
              Ноябрь
            </div>
          </div>
          <div
            class="c25"
            data-cell-type="monthCell"
            data-value="Sun, 01 Dec 2024 09:00:00 GMT"
          >
            <div
              class="c18"
            />
            <div
              class="c16"
            />
            <div
              class="c26"
            >
              Декабрь
            </div>
          </div>
        </div>
      </div>
      <div
        class="c28 c29"
        data-container-type="yearsOfYearWrapper"
      >
        <div
          class="c30"
          data-container-type="yearsWrapper"
        >
          <div
            class="c31"
            data-cell-type="yearCell"
            data-value="Fri, 01 Jan 2021 09:00:00 GMT"
          >
            <div
              class="c18"
            />
            <div
              class="c16"
            />
            <div
              class="c26"
            >
              2021
            </div>
          </div>
          <div
            class="c31"
            data-cell-type="yearCell"
            data-value="Sat, 01 Jan 2022 09:00:00 GMT"
          >
            <div
              class="c18"
            />
            <div
              class="c16"
            />
            <div
              class="c26"
            >
              2022
            </div>
          </div>
          <div
            class="c31"
            data-cell-type="yearCell"
            data-value="Sun, 01 Jan 2023 09:00:00 GMT"
          >
            <div
              class="c18"
            />
            <div
              class="c16"
            />
            <div
              class="c26"
            >
              2023
            </div>
          </div>
          <div
            class="c31"
            data-cell-type="yearCell"
            data-is-current-year="true"
            data-value="Mon, 01 Jan 2024 09:00:00 GMT"
          >
            <div
              class="c18"
            />
            <div
              class="c16"
            />
            <div
              class="c27"
            >
              2024
            </div>
          </div>
          <div
            class="c31"
            data-cell-type="yearCell"
            data-value="Wed, 01 Jan 2025 09:00:00 GMT"
          >
            <div
              class="c18"
            />
            <div
              class="c16"
            />
            <div
              class="c26"
            >
              2025
            </div>
          </div>
          <div
            class="c31"
            data-cell-type="yearCell"
            data-value="Thu, 01 Jan 2026 09:00:00 GMT"
          >
            <div
              class="c18"
            />
            <div
              class="c16"
            />
            <div
              class="c26"
            >
              2026
            </div>
          </div>
          <div
            class="c31"
            data-cell-type="yearCell"
            data-value="Fri, 01 Jan 2027 09:00:00 GMT"
          >
            <div
              class="c18"
            />
            <div
              class="c16"
            />
            <div
              class="c26"
            >
              2027
            </div>
          </div>
          <div
            class="c31"
            data-cell-type="yearCell"
            data-value="Sat, 01 Jan 2028 09:00:00 GMT"
          >
            <div
              class="c18"
            />
            <div
              class="c16"
            />
            <div
              class="c26"
            >
              2028
            </div>
          </div>
          <div
            class="c31"
            data-cell-type="yearCell"
            data-value="Mon, 01 Jan 2029 09:00:00 GMT"
          >
            <div
              class="c18"
            />
            <div
              class="c16"
            />
            <div
              class="c26"
            >
              2029
            </div>
          </div>
          <div
            class="c31"
            data-cell-type="yearCell"
            data-value="Tue, 01 Jan 2030 09:00:00 GMT"
          >
            <div
              class="c18"
            />
            <div
              class="c16"
            />
            <div
              class="c26"
            >
              2030
            </div>
          </div>
          <div
            class="c31"
            data-cell-type="yearCell"
            data-value="Wed, 01 Jan 2031 09:00:00 GMT"
          >
            <div
              class="c18"
            />
            <div
              class="c16"
            />
            <div
              class="c26"
            >
              2031
            </div>
          </div>
          <div
            class="c31"
            data-cell-type="yearCell"
            data-value="Thu, 01 Jan 2032 09:00:00 GMT"
          >
            <div
              class="c18"
            />
            <div
              class="c16"
            />
            <div
              class="c26"
            >
              2032
            </div>
          </div>
          <div
            class="c31"
            data-cell-type="yearCell"
            data-value="Sat, 01 Jan 2033 09:00:00 GMT"
          >
            <div
              class="c18"
            />
            <div
              class="c16"
            />
            <div
              class="c26"
            >
              2033
            </div>
          </div>
          <div
            class="c31"
            data-cell-type="yearCell"
            data-value="Sun, 01 Jan 2034 09:00:00 GMT"
          >
            <div
              class="c18"
            />
            <div
              class="c16"
            />
            <div
              class="c26"
            >
              2034
            </div>
          </div>
          <div
            class="c31"
            data-cell-type="yearCell"
            data-value="Mon, 01 Jan 2035 09:00:00 GMT"
          >
            <div
              class="c18"
            />
            <div
              class="c16"
            />
            <div
              class="c26"
            >
              2035
            </div>
          </div>
          <div
            class="c31"
            data-cell-type="yearCell"
            data-value="Tue, 01 Jan 2036 09:00:00 GMT"
          >
            <div
              class="c18"
            />
            <div
              class="c16"
            />
            <div
              class="c26"
            >
              2036
            </div>
          </div>
          <div
            class="c31"
            data-cell-type="yearCell"
            data-value="Thu, 01 Jan 2037 09:00:00 GMT"
          >
            <div
              class="c18"
            />
            <div
              class="c16"
            />
            <div
              class="c26"
            >
              2037
            </div>
          </div>
          <div
            class="c31"
            data-cell-type="yearCell"
            data-value="Fri, 01 Jan 2038 09:00:00 GMT"
          >
            <div
              class="c18"
            />
            <div
              class="c16"
            />
            <div
              class="c26"
            >
              2038
            </div>
          </div>
          <div
            class="c31"
            data-cell-type="yearCell"
            data-value="Sat, 01 Jan 2039 09:00:00 GMT"
          >
            <div
              class="c18"
            />
            <div
              class="c16"
            />
            <div
              class="c26"
            >
              2039
            </div>
          </div>
          <div
            class="c31"
            data-cell-type="yearCell"
            data-value="Sun, 01 Jan 2040 09:00:00 GMT"
          >
            <div
              class="c18"
            />
            <div
              class="c16"
            />
            <div
              class="c26"
            >
              2040
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
